lib/features/composer/presentation/composer_controller.dart

```diff
@@ -1408,10 +1422,61 @@ class ComposerController extends BaseController with DragDropFileMixin implement
     _initAttachmentsAndInlineImages(
       attachments: success.attachments,
       inlineImages: success.inlineImages);
+
+    if (composerArguments.value?.emailActionType == EmailActionType.editDraft) {
+      success = _hideSignature(success);
+    }
+
     emailContentsViewState.value = Right(success);
 
     if (composerArguments.value?.emailActionType == EmailActionType.editDraft) {
       _setUpRequestReadReceiptForDraftEmail(success.emailCurrent);
+      _restoreIdentityFromHeader(success.emailCurrent);
+    }
+  }
+
+  void _restoreIdentityFromHeader(Email? email) {
+    final identityIdFromHeader = email?.identityIdFromHeader;
+    if (identityIdFromHeader == null) return;
+    final selectedIdentityFromHeader = _selectIdentityFromId(identityIdFromHeader);
+    if (selectedIdentityFromHeader == null) return;
+    identitySelected.value = selectedIdentityFromHeader;
+  }
+
+  Future<void> restoreCollapsibleButton(String? emailContent) async {
+    try {
+      if (emailContent == null) return;
+      final emailDocument = parse(emailContent);
+      final signature = emailDocument.querySelector('div.tmail-signature');
+      if (signature == null) return;
+      await _applySignature(signature.innerHtml);
+    } catch (e) {
+      logError('ComposerController::_restoreCollapsibleButton: $e');
+    }
+  }
+
+  GetEmailContentSuccess _hideSignature(GetEmailContentSuccess success) {
+    try {
+      final htmlContent = success.htmlEmailContent;
+      final emailDocument = parse(htmlContent);
+      final signature = emailDocument.querySelector('div.tmail-signature');
+      if (signature == null) return success;
+      final currentStyle = signature.attributes['style']?.trim();
+      if (currentStyle == null) {
+        signature.attributes['style'] = 'display: none;';
+      } else if (currentStyle.endsWith(';')) {
+        signature.attributes['style'] = '$currentStyle display: none;';
+      } else {
+        signature.attributes['style'] = '$currentStyle; display: none;';
+      }
+      return GetEmailContentSuccess(
+        htmlEmailContent: emailDocument.outerHtml,
+        emailCurrent: success.emailCurrent,
+        attachments: success.attachments,
+        inlineImages: success.inlineImages);
+    } catch (e) {
+      logError('ComposerController::_hideSignature: $e');
+      return success;
     }
```

Comment:
Is there a way to push this processing down to the data layer?.

Reply:
Done